### PR TITLE
feat(docs): switch themes functionality

### PR DESCRIPTION
**Pull request by ti10le**
#1034 

- initial change to marigold-theme if we have one
- first version, please review and I will make it better :)

### CODE REVIEW

File: docs/src/components/ThemeSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import b2bTheme from '@marigold/theme-b2b';
+
+export const themes = {
+  b2bTheme,
+  unicornTheme,
+};
+
+const themeSelectStyles = {
+  position: 'absolute',
+  right: b2bTheme.space.none,
```
Comment (sebald) [inline docs/src/components/ThemeSelect.tsx:14]: That's not how this works :-O Use the scaling values! :P

File: docs/src/components/ThemeSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  cursor: 'pointer',
+  fontSize: b2bTheme.fontSizes.xxsmall,
+  fontFamily: b2bTheme.fonts.body,
+};
+
+export const ThemeSelect: React.FC = () => {
+  const { current, themes, setTheme } = useThemeSwitch();
+
+  return (
+    <select
```
Comment (sebald) [inline docs/src/components/ThemeSelect.tsx:30]: Why are you not using Marigold's `<Select>` component?

File: docs/src/components/ThemeSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { useThemeSwitch } from './ThemeSwitch';
+import { useStyles } from '@marigold/system';
+import unicornTheme from '@marigold/theme-unicorn';
+import b2bTheme from '@marigold/theme-b2b';
+
+export const themes = {
```
Comment (sebald) [inline docs/src/components/ThemeSelect.tsx:7]: You are using the same var name that is returned from the hook.

File: docs/src/components/CodeBlock.tsx [hunk 4]
```diff
@@ -71,7 +72,9 @@ export const CodeBlock: React.FC<CodeBlockProps> = ({
                     <LivePreview />
                   </LiveProvider>
                 </div>
-                <ShowHideButton hide={hide} onHideChange={setHide} />
+                <ThemeProvider theme={b2bTheme}>
```
Comment (sebald) [inline docs/src/components/CodeBlock.tsx:75]: Shouldn't the switch theme stuff be here?

**Review by sebald — CHANGES_REQUESTED** [review]

File: docs/src/components/CodeBlock.tsx (hunk 4 above)
Comment (ti10le) [reply docs/src/components/CodeBlock.tsx:75]: Then you have the problem that you have to change the theme for every component example separately. Copy/show button should have the initial theme

File: docs/src/components/ThemeSelect.tsx (hunk 2 above)
Comment (ti10le) [reply docs/src/components/ThemeSelect.tsx:30]: Our Select doesn't like it should. We have to rewrite some things. Is this e separate issue our should I do this here?

File: docs/src/components/ThemeSelect.tsx (hunk 1 above)
Comment (ti10le) [reply docs/src/components/ThemeSelect.tsx:14]: I am using normal css values until we have a marigold-theme.

File: docs/src/components/ThemeSelect.tsx (hunk 2 above)
Comment (sebald) [reply docs/src/components/ThemeSelect.tsx:29]: That's why we are dogfooding! I've done the same thing #1132 ... I wanted to use our `<Link>` component but it didn't work. That's our chance to fix this!

File: docs/src/components/CodeBlock.tsx (hunk 4 above)
Comment (sebald) [reply docs/src/components/CodeBlock.tsx:75]: Not quite sure I follow. You can add the switch globally and then pass it to the `<ThemeProvider>` inside the `<Codeblock!>` :)

File: docs/src/components/ThemeSwitch.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+export const useThemeSwitch = () => useContext(Context);
+
+// Component
+// ---------------
+export type MarigoldThemeSwitchProps = {
+  themes: { [name: string]: BaseTheme };
+  initial?: string;
+};
+
+export const MarigoldThemeSwitch: React.FC<MarigoldThemeSwitchProps> = ({
```
Comment (sebald) [inline docs/src/components/ThemeSwitch.tsx:29]: Separate the switch from the `<ThemeProvider>`.

File: docs/src/components/ThemeSelect.tsx (hunk 2 above)
Comment (ti10le) [reply docs/src/components/ThemeSelect.tsx]: New issue/PR to rewrite the Select component

**Comment by sebald** [comment]
Great work!

**Review by sebald — APPROVED** [review]

**Pull request merged**